src/Api/Controllers/AccountsController.cs

```diff
@@ -467,14 +467,19 @@ public async Task<PaymentResponseModel> PostPremium(PremiumRequestModel model)
             {
                 license = await ApiHelpers.ReadJsonFileFromBody<UserLicense>(HttpContext, model.License);
             }
+            else if (!valid && !_globalSettings.SelfHosted)
+            {
+                throw new BadRequestException("Country is required.");
+            }
 
             if (!valid || (_globalSettings.SelfHosted && license == null))
             {
                 throw new BadRequestException("Invalid license.");
             }
 
             var result = await _userService.SignUpPremiumAsync(user, model.PaymentToken,
-                model.PaymentMethodType.Value, model.AdditionalStorageGb.GetValueOrDefault(0), license);
+                model.PaymentMethodType.Value, model.AdditionalStorageGb.GetValueOrDefault(0), license,
+                model.Country, model.PostalCode);
```

Comment:
Maybe pass TaxInfo instead?

Reply:
I can do that for consistency sake, seemed overkill since we only need those 2 fields for premium users.